core/vm/interpreter.go

```diff
@@ -249,12 +249,15 @@ func (in *EVMInterpreter) Run(contract *Contract, input []byte, readOnly bool) (
 			if err != nil || !contract.UseGas(dynamicCost) {
 				return nil, ErrOutOfGas
 			}
-		}
-		if memorySize > 0 {
-			mem.Resize(memorySize)
-		}
```

Comment:
Is there any reason why we did `mem.Resize` outside the `dynamicGas` if branch previously? Or it's just legacy?

Reply:
old codes from upstream.  The problem is also reported by us last year. (No team like us cares the correctness & consistency of traces so much since we rebuild states and align each edge case behavior with evm..)